float32 rho
float32 theta
float32 speed
float32 K
bool open
---
int32 err